src/Installers/Windows/HostOptions/Product.wxs

```diff
@@ -0,0 +1,82 @@
+        <Media Id="1" />
+
+        <!-- Custom Actions to assign a default of 0 to all switches -->
+        <CustomAction Id="Set_OPT_NO_ANCM" Property="OPT_NO_ANCM" Value="0" />
+        <CustomAction Id="Set_OPT_NO_FTS" Property="OPT_NO_FTS" Value="0" />
+        <CustomAction Id="Set_OPT_NO_LTS" Property="OPT_NO_LTS" Value="0" />
+        <CustomAction Id="Set_OPT_NO_SHAREDFX" Property="OPT_NO_SHAREDFX" Value="0" />
+        <CustomAction Id="Set_OPT_NO_RUNTIME" Property="OPT_NO_RUNTIME" Value="0" />
+        <CustomAction Id="Set_OPT_NO_X86" Property="OPT_NO_X86" Value="0" />
+        <CustomAction Id="Set_OPT_NO_SHARED_CONFIG_CHECK" Property="OPT_NO_SHARED_CONFIG_CHECK" Value="0" />
+
+        <InstallExecuteSequence>
+            <!-- Only set the options to 0 if they weren't already set in the hosting bundle. -->
+            <!-- First option is to use user input, if they passed any (if they explicitly set an option to the empty string, we convert that to 0 here). -->
+            <!-- Second option is to use registry values, if present. -->
+            <!-- Third option is to set the options all to 0, which we do here. -->
+            <Custom Action="Set_OPT_NO_ANCM" After="AppSearch">NOT OPT_NO_ANCM OR OPT_NO_ANCM=""</Custom>
+            <Custom Action="Set_OPT_NO_FTS" After="AppSearch">NOT OPT_NO_FTS OR OPT_NO_FTS=""</Custom>
+            <Custom Action="Set_OPT_NO_LTS" After="AppSearch">NOT OPT_NO_LTS OR OPT_NO_LTS=""</Custom>
+            <Custom Action="Set_OPT_NO_SHAREDFX" After="AppSearch">NOT OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=""</Custom>
+            <Custom Action="Set_OPT_NO_RUNTIME" After="AppSearch">NOT OPT_NO_RUNTIME OR OPT_NO_RUNTIME=""</Custom>
+            <Custom Action="Set_OPT_NO_X86" After="AppSearch">NOT OPT_NO_X86 OR OPT_NO_X86=""</Custom>
+            <Custom Action="Set_OPT_NO_SHARED_CONFIG_CHECK" After="AppSearch">NOT OPT_NO_SHARED_CONFIG_CHECK OR OPT_NO_SHARED_CONFIG_CHECK=""</Custom>
+        </InstallExecuteSequence>
+
+        <Feature Id="ProductFeature" Title="HostOptions" Level="1">
+            <ComponentGroupRef Id="ProductComponents" />
+        </Feature>
+    </Product>
+
+    <Fragment>
+        <Directory Id="TARGETDIR" Name="SourceDir">
+            <Directory Id="ProgramFilesFolder">
+                <Directory Id="INSTALLFOLDER" Name="dotnet">
+                    <Directory Id="MM" Name="$(var.MajorVersion).$(var.MinorVersion)" />
+                </Directory>
+            </Directory>
+        </Directory>
+    </Fragment>
+
+    <Fragment>
+        <ComponentGroup Id="ProductComponents" Directory="MM">
+
+            <?ifdef ProductOptionsKey?>
+            <?undef ProductOptionsKey?>
+            <?endif?>
+
+            <?define ProductOptionsKey=SOFTWARE\Microsoft\dotnet\host\options\$(var.MajorVersion).$(var.MinorVersion)?>
+
+            <Component Id="OPT" KeyPath="yes" Guid="{5FF00E0C-2BEE-42CA-8359-D46D6CB74DD1}">
```

Comment:
Remove the KeyPath=yes and set `Guid="*"`. then on the RegistryValue, set KeyPath="yes". This way, the compiler will compute stable GUIDs. Static GUIDs usually end up causing issues, eg. if code is copied around and someone forgets to change them. If you collapse the registry values, you only need to set the keypath on the first value. The downside of that is that if any of the other values are absent, repair won't fix them. The alternative to that is to move the registry values into a separate component - that would be more verbose but provides the most flexibility and works better with repair operations